ipn/localapi/localapi.go

```diff
@@ -944,13 +941,13 @@ func authorizeServeConfigForGOOSAndUserContext(goos string, configIn *ipn.ServeC
 	if !slices.Contains([]string{"windows", "linux", "darwin"}, goos) {
 		return nil
 	}
-	if goos == "darwin" && !version.IsSandboxedMacOS() {
+	if goos == "darwin" && version.IsSandboxedMacOS() {
 		return nil
```

Comment:
if this code was confused before, people will likely be confused in the future about why. Add a little comment in this block explaining why sandboxed (GUI) is special?